### PR TITLE
Updating parallel tutorial pages

### PR DESCRIPTION
**Pull request by simo-tuomisto**
This PR updates the parallel jobs tutorial page with various improvements.

### CODE REVIEW

File: triton/tut/parallel.rst [hunk 1]
```diff
@@ -3,54 +3,78 @@ Parallel computing
...
-  Examples of application that utilize this model: Matlab, R, OpenMP
-  applications, typical parallel desktop programs.
+* Shared memory (or multithreaded) programs run multiple independent
+  workers on the same machine. As the name suggests, all of the computer's
+  memory has to be accessible to all of the processes. **Thus programs
+  that utilize this model are limited to one node only.** Likewise, the
+  maximum number of workers is usually the number of CPU cores available
+  on the computational node. The code is easier to implement and the same
+  code can still be run in a serial mode. Example applications that
+  utilize this model: Matlab, R, Python multithreading/multiprocessing,
```
Comment (rkdarst) [inline triton/tut/parallel.rst:28]:
```suggestion
  utilize this model: Matlab, R, Python multithreading (though not actually the `multiprocessing` module, but it is run a similar way),
```
Maybe this should be glossed over?

File: triton/tut/parallel.rst [hunk 2]
```diff
@@ -77,16 +101,26 @@ node regardless of number of processors), use ``--mem``.
...
-  #SBATCH --cpus-per-task=12
+  #SBATCH --cpus-per-task=4
   #SBATCH  --mem-per-cpu=2000
-  #SBATCH --time=45:00
+  #SBATCH --time=00:30:00
+
   export OMP_PROC_BIND=TRUE
   srun omp_program
 
 Good to know that OpenMP is both an environment and set of libraries, but
```
Comment (rkdarst) [inline triton/tut/parallel.rst:111]:
```suggestion
It is good to know that OpenMP is both an environment and set of libraries, but
```

File: triton/tut/parallel.rst [hunk 3]
```diff
@@ -3,54 +3,78 @@ Parallel computing
...
+* Shared memory (or multithreaded) programs run multiple independent
+  workers on the same machine. As the name suggests, all of the computer's
+  memory has to be accessible to all of the processes. **Thus programs
+  that utilize this model are limited to one node only.** Likewise, the
+  maximum number of workers is usually the number of CPU cores available
+  on the computational node. The code is easier to implement and the same
+  code can still be run in a serial mode. Example applications that
+  utilize this model: Matlab, R, Python multithreading/multiprocessing,
+  OpenMP applications, BLAS libraries, FFTW libraries, typical
+  multithreaded parallel pesktop programs.
```
Comment (rkdarst) [inline triton/tut/parallel.rst:30]:
```suggestion
  multithreaded parallel pesktop programs.  **For OpenMP programs, you
  request one node with multiple processors, and start one process which
  is told to use all processors)**
```
Shall we introduce this concept early?

File: triton/tut/parallel.rst [hunk 4]
```diff
@@ -3,54 +3,78 @@ Parallel computing
...
 * Message passing programming (e.g. MPI, message passing interface)
   can run on multiple nodes interconnected with the network via passing
-  data through MPI software libraries. The large-scale scientific programs
-  are MPI. MPI can scale to thousands of CPU cores, but it's harder to
-  implement from the programmer point of view.
+  data through MPI software libraries. Almost all large-scale scientific
+  programs utilize MPI. MPI can scale to thousands of CPU cores, but
+  depending on the case it can be harder to implement from the
+  programmer's point of view. Example applications that utilize this
+  model: CP2K, GPAW, LAMMPS, OpenFoam.
```
Comment (rkdarst) [inline triton/tut/parallel.rst:38]:
```suggestion
  model: CP2K, GPAW, LAMMPS, OpenFoam.  **For MPI programs, you
  request one/multiple nodes and multiple processors, and usually start one
  process for each processor - and they figure out how to communicate themselves.**
```

File: triton/tut/parallel.rst (hunk 1 above)
Comment (simo-tuomisto) [reply triton/tut/parallel.rst:28]: There's a difference in multithreading and multiprocessing in Python (especially with regards to the GIL), but as they both are run using the same way, I would not make a distinction here. Matlab and R use both multithreading and multiprocessing as well.

I would keep that the way it was.

File: triton/tut/parallel.rst (hunk 3 above)
Comment (simo-tuomisto) [reply triton/tut/parallel.rst:30]: I'll write this in a bit different way.

File: triton/tut/parallel.rst (hunk 1 above)
Comment (simo-tuomisto) [reply triton/tut/parallel.rst]: I wrote more information on multithreaded / multiprocessing. This is no longer relevant.

File: triton/tut/parallel.rst (hunk 3 above)
Comment (simo-tuomisto) [reply triton/tut/parallel.rst]: Also fixed in the current version.

File: triton/tut/parallel.rst (hunk 4 above)
Comment (simo-tuomisto) [reply triton/tut/parallel.rst:41]: I added this information in another example.

**Comment by simo-tuomisto** [comment]
This is now IMO ready for production.

**Comment by rkdarst** [comment]
I looked at it a bit ago, I'd say it looks good.  Thanks!

There's a lot of text, but it serves as both an intro and how-to.  I think it's good, but keep in mind when teaching what you will focus on and not - summarize when teaching, they read it fully when they need.

**Pull request merged**